### ISSUE TITLE
should an window.open() about:blank document be controlled?

### ISSUE THREAD

**Issue by wanderview** [w3c/ServiceWorker#1163]
Over in #612 we decided that about:blank documents in iframes should inherit their controller from the parent.  This matches the inheritence of the origin.

I think we also similarly inherit the origin when creating the about:blank document for `window.open()`.  Should that about:blank client also be controlled by the opener's service workers?

**Comment by jakearchibald** [comment]
Repeated from IRC:

I don't think many developers understand about:blank's relationship to the parent/opener. But I think making iframe about:blank and window.open about:blank behave the same is the best thing.

**Comment by jungkees** [comment]
Agreed. The relationship between the auxilary browsing context and the createor browsing context is defined as the same as that of between the child and the parent. So, I think maintaining consistent behaviors make sense.

**Comment by yoshisatoyanagisawa** [comment]
Triage: the request sounds valid, and we need to update the spec.

**Comment by asutherland** [comment]
Do we think the (post-2017 text) at https://w3c.github.io/ServiceWorker/#control-and-use-window-client doesn't cover this already?  I think I had read it as covering this when I'd most recently read, but I also was reading assuming the inheriting happened.  (cc gecko about:blank expert @vinhill)

Aside: Gecko definitely has implemented the inheriting of the controller from a controlled window using window.open() into the opened about:blank.  (That said, it has been a source of frustration in terms of anti-tracking heuristics impacting partitioning since the resulting browsing context is at once top-level as well as being an auxiliary browsing context/having an opener.  (xref https://github.com/whatwg/html/issues/5680)

**Comment by yoshisatoyanagisawa** [comment]
As Andrew pointed out, this behavior appears to be covered in Section 2.5.1, 'The window client case.' However, to avoid confusion, adding a note specifically for `about:blank` to clarify this inheritance might be beneficial.

Also, following up on Andrew's comment, we may want to re-evaluate the implications for partitioning and anti-tracking. I would like to hear others' thoughts on these aspects. If the current logic is deemed acceptable as-is, let’s proceed with adding the clarifying note.

**Comment by asutherland** [comment]
Unfortunately the partitioning/anti-tracking case is a real mess in Gecko right now.  Our anti-tracking code currently wants the auxiliary browsing context to not be partitioned even as it's inheriting a partitioned controller.  We're still working through this right now.

**Comment by asutherland** [comment]
Okay, so I think in Gecko it's untenable for us to inherit the controller into auxiliary browsing contexts via window.open().  It turns out we have **not** been inheriting in this auxiliary browsing context case, but in https://bugzilla.mozilla.org/show_bug.cgi?id=543435 on beta we tried to start doing it because it was our reading of the spec text that we should (and honestly I thought we'd been inheriting it), but it's effectively incompatible with our storage partitioning implementation which thinks that such a window should not be partitioned[1].  We are planning to remove the inheriting again before it reaches release.

Our anti-tracking team has indicated that they think there are potentially significant webcompat problems with attempting to partition in the auxiliary browsing context case, so it is unlikely for this to change any time soon.

So I guess Gecko's answer to the question "should an window.open() about:blank document be controlled by opener's controller" is now no and I think we'd like to get consensus where other browsers are on this, especially those that implement partitioning.

cc @annevk @youennf

1: That said, we could do a ton of carve-outs so that the about:blank in fact does inherit, but it would be very contradictory since then a navigation to a same-origin (non-about:blank) URL would go from being partitioned to being unpartitioned and we would potentially be fighting a ton of invariant assertions, which is our current problem because a bunch of safety checks can get very upset.
